project/dataset/valdo.py

```diff
@@ -55,4 +57,138 @@ def count_cmb_per_image(self, segmented_images):
+
+    #     :param idx: Index of the MRI scan in the dataset.
+    #     :param x_min: Minimum x coordinate of the location.
+    #     :param y_min: Minimum y coordinate of the location.
+    #     :param x_max: Maximum x coordinate of the location.
+    #     :param y_max: Maximum y coordinate of the location.
+    #     """
+    #     try:
+    #         # img_path = self.img_paths[idx]
+
+    #         # # Load 3D image
+    #         # img = nib.load(img_path).get_fdata()
+    #         # # img = (img / np.max(img) * 255).astype(np.uint8)
+    #         # img, targets, img_path, cmb_count = self.__getitem__(idx=idx)
+    #         cropped_slices = []
+    #         max_width = 3072
+
+    #         # Iterate through each slice and crop to the specified region
+    #         # print(img.shape)
+    #         if img.dim() == 5:
+    #             # If 5D, assume shape is [num_slices, channels, height, width]
+    #             for i in range(img.__len__()):
+    #                 img_slice = img[i]
+    #                 cropped_slice = img_slice[0, 0, y_min:y_max, x_min:x_max]
+    #                 cropped_slices.append(torch.Tensor(cropped_slice))
+
+    #             # combined_slices = np.hstack(cropped_slices)
+
+    #             # current_width = combined_slices.shape[1]
+    #             # pad_width = max_width - current_width
+    #             # padded_image = np.pad(combined_slices, ((0, 0), (0, pad_width)), mode='constant')
+
+    #             # plt.imshow(padded_image, cmap='gray')
+    #             # plt.title(f'Slice {i}')
+    #             # plt.show()
+    #             return cropped_slices
+
+    #         elif img.dim() == 4:
+    #             # If 4D, assume shape is [num_slices, height, width]
+    #             for i in range(img.shape[0]):
+    #                 img_slice = img[i, 0]  # Extract the 2D slice (assuming single channel)
+    #                 cropped_slice = img_slice[y_min:y_max, x_min:x_max]
+    #                 cropped_slices.append(torch.Tensor(cropped_slice))
+                    
+    #                 # for i in range(img.__len__()):
+    #                 #     returned.append(cropped_slices)
+    #             # combined_slices = np.hstack(cropped_slices)
+    #             # current_width = combined_slices.shape[1]
+    #             # pad_width = max_width - current_width
+    #             # padded_image = np.pad(combined_slices, ((0, 0), (0, pad_width)), mode='constant')
+
+    #             # plt.imshow(padded_image, cmap='gray')
+    #             # plt.title(f'Slice {i}')
+    #             # plt.show()
+    #             return cropped_slices
+
+    #         else:
+    #             raise ValueError("Unsupported tensor dimension. Expected 3D or 4D tensor.")
+
+
+    #     except Exception as e:
+    #         print(f"Error processing: {e}")
+    #         raise
+    def get_cropped_locations(self, img, x_min, y_min, x_max, y_max):
```

Comment:
What's the use of this?

Reply:
gawa yan ni lex during sa development ng original lumang vit pinapull nya ako kaya na update yung function. pero yan yugn ginagamit ko ngayon.

Reply:
Pwede ko ba tanggalin to or kailangan mo?

Reply:
tanggalin mo lang yung mga naka comment out

Reply:
Sige retain ko si `get_cropped_locations`